nautilus_trader/data/aggregation.pyx

```diff
@@ -101,23 +101,24 @@ cdef class BarBuilder:
             The partial bar with values to set.
 
         """
-        if self._partial_set:
+        if self._partial_set and run_once:
             return  # Already updated
 
-        self._open = partial_bar.open
+        if self._open is None:
+            self._open = partial_bar.open
 
         if self._high is None or partial_bar.high > self._high:
             self._high = partial_bar.high
 
         if self._low is None or partial_bar.low < self._low:
             self._low = partial_bar.low
 
-        if self._close is None:
+        if self._close is None or not run_once:
             self._close = partial_bar.close
 
-        self.volume = partial_bar.volume
+        self.volume = Quantity(self.volume + partial_bar.volume, self.size_precision)
```

Comment:
Could do an `add_assign` too. The precisions should be the same if we enforce that the instrument IDs have to be the same?

Reply:
Hi Chris, for this two quantities are equal to a decimal, that's why I needed to do this